File: files/problems/js/fizz-buzz.js
// by @codeAbinash
// Time Complexity : O(n)
// Space Complexity : O(1)

var fizzBuzz = function (n) {
    arr = []
    for (let i = 1; i <= n; i++) {
        if (i % 3 == 0 && i % 5 == 0)
            arr.push("FizzBuzz")
        else if (i % 3 == 0)
            arr.push("Fizz")
        else if (i % 5 == 0)
            arr.push("Buzz")
        else
            arr.push(i + "")
    }
    return arr
};

File: leetcode/problems/cpp/fizz-buzz.cpp
// by @codeAbinash
// Time : O(n)
// Space : O(1)


#include "iostream"
#include "vector"
using namespace std;


class Solution {
public:
  vector<string> fizzBuzz(int n) {
    vector<string> ans;
    for (int i = 1; i <= n; i++) {
      if (i % 3 == 0 && i % 5 == 0)
        ans.push_back("FizzBuzz");
      else if (i % 3 == 0)
        ans.push_back("Fizz");
      else if (i % 5 == 0)
        ans.push_back("Buzz");
      else ans.push_back(to_string(i));
    }
    return ans;
  }
};
